src/transformers/models/detr/configuration_detr.py

```diff
@@ -169,6 +179,18 @@ def __init__(
         eos_coefficient=0.1,
         **kwargs
     ):
+        if not use_timm_backbone:
+            if backbone_config is None:
+                logger.info("`backbone_config` is `None`. Initializing the config with the default `ResNet` backbone.")
+                backbone_config = ResNetConfig(out_features=["stage2", "stage3", "stage4"])
+            else:
+                if isinstance(backbone_config, dict):
+                    backbone_model_type = backbone_config.pop("model_type")
```

Comment:
This will modify the input dictionary - I would just use `get` or `backbone_config['model_type']` depending on whether you want an issue to be raised.